### PR TITLE
update security best practice

### PR DESCRIPTION
**Pull request by pmmgithub**
as requested by PM to align with RTC

### CODE REVIEW

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md [hunk 1]
```diff
@@ -24,26 +55,18 @@ Agora Chat's data center locations and corresponding service areas are listed as
...
 | North America  | Virginia, US       | North America  |
 
-To use Agora Chat, you need to specify a data center. After you select a data center, both the REST requests and the SDK API requests to the message server are directed to the data center accordingly.
-Once selected, the data center cannot be changed. Agora Chat does not support data migration across service areas. All data is stored in the designated data centers.
+To use Chat, you need to specify a data center. After you select a data center, both the RESTful requests and the SDK API requests to the message server are directed to the data center accordingly.
+Once selected, the data center cannot be changed. Chat does not support data migration across service areas. All data is stored in the designated data centers.
 
-## Authentication with tokens
+## Security best practice checklist
+Use this list to quickly check what protection measures you need to take to ensure the security of you app and users:
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:62]: Edit: "your app" (however, this list isn't the checklist you seem to be introducing? Maybe move this section under ## Level 2 - Token authentication?) Or rewrite the list?

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md [hunk 2]
```diff
@@ -24,26 +55,18 @@ Agora Chat's data center locations and corresponding service areas are listed as
 | Europe         | Frankfurt, Germany | Europe         |
 | North America  | Virginia, US       | North America  |
 
-To use Agora Chat, you need to specify a data center. After you select a data center, both the REST requests and the SDK API requests to the message server are directed to the data center accordingly.
-Once selected, the data center cannot be changed. Agora Chat does not support data migration across service areas. All data is stored in the designated data centers.
+To use Chat, you need to specify a data center. After you select a data center, both the RESTful requests and the SDK API requests to the message server are directed to the data center accordingly.
+Once selected, the data center cannot be changed. Chat does not support data migration across service areas. All data is stored in the designated data centers.
 
-## Authentication with tokens
+## Security best practice checklist
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:61]: This really is a list of how to set up token authentication, not a "best practice checklist". I recommend going back to the original heading.

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md [hunk 3]
```diff
@@ -24,26 +55,18 @@ Agora Chat's data center locations and corresponding service areas are listed as
...
+Use this list to quickly check what protection measures you need to take to ensure the security of you app and users:
 
-Agora Chat uses tokens to validate the identities of end users. A token is a dynamic key  generated by the application client with a validation period set by the developers. Tokens ensure that only authenticated users have access to Agora Chat. Each token contains the following information:
+1. Enable token authentication on [Agora Console](https://console.agora.io/).
+2. Disable **No certificate** on your project management page so that your app authenticates users with tokens only.
+3. [Deploy a token server](https://docs.agora.io/en/agora-chat/develop/generate-user-tokens#deploy-a-token-server) in your backend services.
+4. Protect the token server by only allowing the app backend server to connect to the token server.
+5. Set the token expiration time to a reasonable value. 
+6. To strengthen security, do not use real user IDs in your app as `username` in Chat. Instead, use randomly-generated IDs as `username`.
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:69]: Edit: "randomly generated" (no need to hyphenate -ly words)

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md [hunk 4]
```diff
@@ -24,26 +55,18 @@ Agora Chat's data center locations and corresponding service areas are listed as
...
 
-## Authentication with tokens
+## Security best practice checklist
+Use this list to quickly check what protection measures you need to take to ensure the security of you app and users:
 
-Agora Chat uses tokens to validate the identities of end users. A token is a dynamic key  generated by the application client with a validation period set by the developers. Tokens ensure that only authenticated users have access to Agora Chat. Each token contains the following information:
+1. Enable token authentication on [Agora Console](https://console.agora.io/).
+2. Disable **No certificate** on your project management page so that your app authenticates users with tokens only.
+3. [Deploy a token server](https://docs.agora.io/en/agora-chat/develop/generate-user-tokens#deploy-a-token-server) in your backend services.
+4. Protect the token server by only allowing the app backend server to connect to the token server.
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:67]: Edit: "app's backend"

**Review by mst3k4ever — COMMENTED** [review]
:)

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md [hunk 5]
```diff
@@ -1,21 +1,52 @@
...
+- Token with user privileges: used to allow users to access the Chat service.
+- Token with app privileges: used to allow app developers to perform administrator-level operations on the app with RESTful APIs.
+
+To generate a token with user privileges, you need to provide unique information, including App ID, App Certificate, `uuid`, and the token expiration time.
+![](https://web-cdn.agora.io/docs-files/1665213460331)
+
+Developers can enable the token authentication system (achieved by App Certificate) on Agora Console. Once enabled, valid tokens are required for all users who want to log into Chat.
+![](https://web-cdn.agora.io/docs-files/1665213588546)
+
+Please note that developers need to set a token expiration time (with a default value of 24 hours) to ensure security. A token contains the timestamp of the expiration time during which users are allowed to use that Chat service.
```
[the remark this answers is not in the document]
Comment (pmmgithub) [reply en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:28]: You are right. The token restricts access, I'll clarify this point. :)

File: en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md (hunk 1 above)
Comment (pmmgithub) [reply en-US/markdown/agora-chat/Reference/agora_chat_security_practice.md:62]: Currently the list only has token-related information, but it's supposed to have other security measures in the future. So I'll keep it as it is for now. :)

**Pull request merged**